fava/util/date.py

```diff
@@ -307,6 +308,8 @@ def get_fiscal_period(year, fye, quarter=None):
                 )
                 + datetime.timedelta(days=1)
             ).date()
+            if fye == "02-28" and calendar.isleap(year - 1):
```

Comment:
This is a bit confusing I find since independently of whether the year is a leap year, we set the date to x-03-01. So let's make that more explicit (and simplify the code a bit) by only making the first check and then setting the date directly

Reply:
That's what I did originally: add a month to the date and then set day to 1.

Unfortunately, that doesn't work because it breaks FYE like 05-06.

Reply:
Oh, I see what you mean.  Sorry, I misunderstood.  Yeah, ok, I can do that